File: java/Fibonacci.java
class Fibonacci {
    public static void main(String[] args) {
        long start = System.nanoTime();
        System.out.println(fibonacci(25));
        System.out.println(System.nanoTime() - start);
    }
    
    public static int fibonacci(final int n) {
        return (n < 2) ? n : fibonacci(n - 1) + fibonacci(n - 2);
    }
}

File: scala/Fibonacci.scala
object Fibonacci {
    def main(args: Array[String]): Unit = {
        val start = System.nanoTime()
        println(fibonacci(25))
        println(System.nanoTime() - start)
    }
    
    def fibonacci(n : Int) : Int = n match {
        case 0 | 1 => n
        case _ => fibonacci(n - 1) + fibonacci(n - 2)
    }
}
